packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js

```diff
@@ -94,177 +96,182 @@ export const EditFolderDialog = ({ onClose, folder, folderStructure, canUpdate }
     }
   };
 
-  const handleClose = () => {
-    onClose();
-  };
-
-  const handleDelete = async event => {
-    event.preventDefault();
-
+  const handleDelete = async () => {
     await remove([folder]);
 
+    setShowConfirmDialog(false);
     onClose();
   };
 
   return (
-    <ModalLayout onClose={handleClose} labelledBy="title">
-      <ModalHeader>
-        <Typography fontWeight="bold" textColor="neutral800" as="h2" id="title">
-          {formatMessage(
-            isEditing
-              ? {
-                  id: getTrad('modal.folder.edit.title'),
-                  defaultMessage: 'Edit folder',
-                }
-              : {
-                  id: getTrad('modal.folder.create.title'),
-                  defaultMessage: 'Add new folder',
-                }
-          )}
-        </Typography>
-      </ModalHeader>
-
-      <ModalBody>
-        <Formik
-          validationSchema={folderSchema}
-          validateOnChange={false}
-          onSubmit={handleSubmit}
-          initialValues={initialFormData}
-        >
-          {({ values, errors, handleChange, setFieldValue }) => (
-            <Form noValidate>
-              <Grid gap={4}>
-                {isEditing && (
-                  <GridItem xs={12} col={12}>
-                    <ContextInfo
-                      blocks={[
-                        {
-                          label: formatMessage({
-                            id: getTrad('modal.folder.create.elements'),
-                            defaultMessage: 'Elements',
-                          }),
-                          value: formatMessage(
-                            {
-                              id: getTrad('modal.folder.elements.count'),
-                              defaultMessage: '{assetCount} assets, {folderCount} folders',
-                            },
-                            {
-                              assetCount: folder?.files?.count ?? 0,
-                              folderCount: folder?.children?.length ?? 0,
-                            }
-                          ),
-                        },
+    <>
```

Comment:
Does that mean we display the delete confirmation overlaying the edit dialog? What do you think about only ever showing one? It would also simplify the component, because it could be handled as an early return:

```js

if (showConfirmDialog) {
  return <Confirm />
}

return ();

```

Reply:
I discussed this with @maevalienard  yesterday to confirm the behavior we wanted - when deleting an asset we have the edit modal + dialog over it - that's why I followed the same pattern. If we change it for the folder we might need to change it for the asset flow too?

I'm not a fan of 2 modals opened either but if we only have one, what happens when the user clicks on cancel inside the confirmation dialog? It might be frustrating for them to have to open the edit modal once again if we already closed it, WDYT?

<img width="1187" alt="image" src="https://user-images.githubusercontent.com/71838159/169541223-090b9f35-1b1d-4c4f-b155-cceceff7b597.png">